src/Azure.Functions.Cli/Kubernetes/KubernetesHelpers.cs

```diff
@@ -170,21 +181,23 @@ internal static string GetKedaResources(string @namespace)
             {
                 int position = 0;
                 var enabledFunctions = nonHttpFunctions.ToDictionary(k => $"AzureFunctionsJobHost__functions__{position++}", v => v.Key);
-                var deployment = GetDeployment(name, @namespace, imageName, pullSecret, minReplicas ?? 0, enabledFunctions);
+                var deployment = GetDeployment(name, @namespace, imageName, pullSecret, minReplicas ?? 0, enabledFunctions, funcAppKeysSecretsCollectionName: funcAppKeysSecretsCollectionName, funcAppKeysConfigMapName: funcAppKeysConfigMapName, mountFuncKeysAsContainerVolume: mountFuncKeysAsContainerVolume);
                 deployments.Add(deployment);
                 scaledobject = GetScaledObject(name, @namespace, triggers, deployment, pollingInterval, cooldownPeriod, minReplicas, maxReplicas);
             }
 
             // Set worker runtime if needed.
-            if (!secrets.ContainsKey(Constants.FunctionsWorkerRuntime))
+            if (!appSettingsSecrets.ContainsKey(Constants.FunctionsWorkerRuntime))
             {
-                secrets[Constants.FunctionsWorkerRuntime] = GlobalCoreToolsSettings.CurrentWorkerRuntime.ToString();
+                appSettingsSecrets[Constants.FunctionsWorkerRuntime] = GlobalCoreToolsSettings.CurrentWorkerRuntime.ToString();
             }
 
-            if (useConfigMap)
+            int resourceIndex = 0;
```

Comment:
I don't think tracking the index is necessary. I just wanted the output to have all dependent resources before the `Deployment` resource, but I don't think it's actually worth introducing a tracking variable. just inserting at 0 should be fine

Reply:
This index is used to insert the Kubernetes resources in the right order, e.g. to create ServiceAccount, Role and RoleBinding resource in the order